### ISSUE TITLE
Filter results with spaces instead of matching them

### ISSUE THREAD

**Issue by Q-u-y** [mattzollinhofer/vue-typeahead-bootstrap#11]
Hi. I'm wondering if you can help me with this. Currently, spaces are matching as literals, and I want to be able to use them as filter instead. Meaning, when user type a space character, instead of matching a space in the name, I want the next character after the space to search from the reduced filtered list,   For example: `cur da` would match `current date`.  Basically, 
Level 1: `cur` would match like normal
+ Level 2: would match from the filtered Level 1
++ Level 3: would match from filtered Level 2 
 ... and so forth, with each space going down a level.

I hope I'm making myself clear. Any help would be appreciated.

**Comment by BozoJim** [comment]
Hmm. This is the second whitespace related issue that has opened recently. Your approach seems a bit complex, is the level thing necessary to fulfill your goal?
My initial thought is to just strip all whitespace from both the input and data set. If I type `cur da` or `curda`, both will match `current date` becauase it's actually evaluated as `currentdate`.

That approach (i think) wouch solve your issue, as well as part of another issue opened recently. #8 

Does that approach achieve your goal? Or am I over-simplifying your request?

**Comment by Q-u-y** [comment]
No. The level thing is just something that popped into my head. It's not necessary programming logic. 

Your logic seems more sound. If we're able to match `cur da` to `current date` and `currently the date is`  with proper highlighting and sorting like we have now, then it's great. How would I go about making the necessary changes?

BTW, thanks for looking into this for me. I really appreciate it.
